File: Programmers/노수일/Q12949.py
def solution(arr1, arr2):
    answer = []

    for i in range(len(arr1)):
        row_arr1 = arr1[i]
        answer_row = []
        for j in range(len(arr2[0])):
            sum_of = 0
            for t in range(len(row_arr1)):
                sum_of += row_arr1[t] * arr2[t][j]
            answer_row.append(sum_of)
        answer.append(answer_row)
    return answer


print(solution([[1, 4], [3, 2], [4, 1]], [[3, 3], [3, 3]]))

File: Programmers/정권호/Q12949.java
package programers.lv2._0627;

public class Q12949 {
    class Solution {
        public int[][] solution(int[][] arr1, int[][] arr2) {
            int[][] answer = {};
            if (arr1[0].length == arr2.length) { // A * B
                answer = new int[arr1.length][arr2[0].length];
                for (int i = 0; i < answer.length; i++) {
                    for (int j = 0; j < answer[i].length; j++) {
                        for (int k = 0; k < arr1[i].length; k++) {
                            answer[i][j] += arr1[i][k] * arr2[k][j];
                        }
                    }
                }
            } else { // B * A
                answer = new int[arr2.length][arr1[0].length];
                for (int i = 0; i < answer.length; i++) {
                    for (int j = 0; j < answer[i].length; j++) {
                        for (int k = 0; k < arr1.length; k++) {
                            answer[i][j] += arr2[i][k] * arr1[k][j];
                        }
                    }
                }
            }

            return answer;
        }
    }
}

